x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlQueryAction.java

```diff
@@ -107,16 +61,16 @@ public RestResponse buildResponse(SqlQueryResponse response) throws Exception {
                 RestResponse restResponse;
 
                 // XContent branch
-                if (xContentType != null) {
-                    XContentBuilder builder = channel.newBuilder(request.getXContentType(), xContentType, true);
+                if (responseMediaType != null && responseMediaType instanceof XContentType) {
```

Comment:
when parsing returns a supertype to `TextFormat` and `XContentType` it makes it harder to implement these two branches.
i think we should probably move the formatting behaviour out of TextFormat. It should only implement methods defined on MediaType (the original XContentType methods)

Reply:
leaving this up to a discussion. With this approach we can have a single parser that returns both XContentType and TextFormat under MediaType interface
XContentType does not implement format method. But maybe these two should have a common createRestResponse method?